pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -289,6 +318,32 @@ func (m *managerImpl) processShutdownEvent() error {
 	klog.V(1).InfoS("Shutdown manager processing shutdown event")
 	activePods := m.getPods()
 
+	defer func() {
+		m.dbusCon.ReleaseInhibitLock(m.inhibitLock)
+		klog.V(1).InfoS("Shutdown manager completed processing shutdown event, node will shutdown shortly")
+	}()
+
+	if m.enableMetrics && m.storage != nil {
+		startTime := time.Now()
+		err := m.storage.Store(state{
```

Comment:
I'm guessing we should emit the timestamps and save the file prior to releasing the inhibitor lock? Otherwise the system may shutdown before we get a chance to write the state.

Reply:
Yes, I've adjusted release inhibitor lock to execute after this.

``` golang
defer func() {
	m.dbusCon.ReleaseInhibitLock(m.inhibitLock)
	klog.V(1).InfoS("Shutdown manager completed processing shutdown event, node will shutdown shortly")
}()
```